packages/graphql/src/components/DirectiveDeclaration.tsx

```diff
@@ -0,0 +1,138 @@
+import {
+  Children,
+  Declaration as CoreDeclaration,
+  List,
+  MemberScope,
+  Name,
+  Refkey,
+  Show,
+} from "@alloy-js/core";
+import { createGraphQLSymbol } from "../symbol-creation.js";
+import { GraphQLMemberScope } from "../symbols/graphql-member-scope.js";
+import { useGraphQLScope } from "../symbols/scopes.js";
+
+/**
+ * Valid directive locations in GraphQL
+ * @see https://spec.graphql.org/October2021/#DirectiveLocation
```

Comment:
We should reference the September 2025 spec https://spec.graphql.org/September2025/#DirectiveLocation